Packages/com.unity.inputsystem/InputSystem/Plugins/InputForUI/InputSystemProvider.cs

```diff
@@ -0,0 +1,679 @@
+            {
+                //TODO: For now it only deals with Keyboard, needs to deal with other devices if we can add bindings
+                //      for Gamepad, etc
+                //TODO: An alternative could be to have an action for next and for previous since shortcut support does
+                //      not work properly
+                if (m_NextPreviousAction.activeControl.device is Keyboard)
+                {
+                    var keyboard = m_NextPreviousAction.activeControl.device as Keyboard;
+                    // Return direction based on whether shift is pressed or not
+                    return keyboard.shiftKey.isPressed ? NavigationEvent.Direction.Previous : NavigationEvent.Direction.Next;
+                }
+            }
+
+            return NavigationEvent.Direction.None;
+        }
+
+        static int SortEvents(Event a, Event b)
+        {
+            return Event.CompareType(a, b);
+        }
+
+        public void OnFocusChanged(bool focus)
+        {
+            m_InputEventPartialProvider.OnFocusChanged(focus);
+        }
+
+        public bool RequestCurrentState(Event.Type type)
+        {
+            if (m_InputEventPartialProvider.RequestCurrentState(type))
+                return true;
+
+            switch (type)
+            {
+                case Event.Type.PointerEvent:
+                {
+                    if (m_TouchState.LastPositionValid)
+                        EventProvider.Dispatch(Event.From(ToPointerStateEvent(m_CurrentTime, m_TouchState, EventSource.Touch)));
+                    if (m_PenState.LastPositionValid)
+                        EventProvider.Dispatch(Event.From(ToPointerStateEvent(m_CurrentTime, m_PenState, EventSource.Pen)));
+                    if (m_MouseState.LastPositionValid)
+                        EventProvider.Dispatch(Event.From(ToPointerStateEvent(m_CurrentTime, m_MouseState, EventSource.Mouse)));
+                    else
+                    {
+                        // TODO maybe it's reasonable to poll and dispatch mouse state here anyway?
+                    }
+
+                    return m_TouchState.LastPositionValid ||
+                        m_PenState.LastPositionValid ||
+                        m_MouseState.LastPositionValid;
+                }
+                // TODO
+                case Event.Type.IMECompositionEvent:
+                //EventProvider.Dispatch(Event.From(ToIMECompositionEvent(currentTime, _compositionString)));
+                //return true;
+                default:
+                    return false;
+            }
+        }
+
+        public uint playerCount => 1; // TODO
+
+        // copied from UIElementsRuntimeUtility.cs
+        static Vector2 ScreenBottomLeftToPanelPosition(Vector2 position, int targetDisplay)
+        {
+            // Flip positions Y axis between input and UITK
+            var screenHeight = Screen.height;
+            if (targetDisplay > 0 && targetDisplay < Display.displays.Length)
```

Comment:
Should this be >=0 or targetDisplay -1 in array index.

Might be fine but I wondered by 0th index never used